<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example 2: Using SEAGLE with Simulated Data • SEAGLE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example 2: Using SEAGLE with Simulated Data">
<meta property="og:description" content="SEAGLE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SEAGLE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/example1.html">Example 1: Using SEAGLE with .txt Input Files</a>
    </li>
    <li>
      <a href="../articles/example2.html">Example 2: Using SEAGLE with Simulated Data</a>
    </li>
    <li>
      <a href="../articles/example3.html">Example 3: Using SEAGLE with GWAS or Next Generation Sequencing Data</a>
    </li>
    <li>
      <a href="../articles/example4.html">Example 4: Using SEAGLE for Chromosome-Wide Gene-Based Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jocelynchi/SEAGLE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="example2_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example 2: Using SEAGLE with Simulated Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jocelynchi/SEAGLE/blob/master/vignettes/example2.Rmd"><code>vignettes/example2.Rmd</code></a></small>
      <div class="hidden name"><code>example2.Rmd</code></div>

    </div>

    
    
<p>This tutorial demonstrates how to use the <code>SEAGLE</code> package when the user inputs a matrix <span class="math inline">\({\bf G}\)</span>. We’ll begin by loading the <code>SEAGLE</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jocelynchi/SEAGLE">SEAGLE</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Loading required package: CompQuadForm</span></code></pre></div>
<p>As an example, we’ll generate some synthetic data for usage in this tutorial. Let’s consider a dataset with <span class="math inline">\(n=5000\)</span> individuals and <span class="math inline">\(L=100\)</span> loci, where the first <span class="math inline">\(40\)</span> are causal.</p>
<p>The <code>makeSimData</code> function generates a covariate matrix <span class="math inline">\(\widetilde{\bf X} \in \mathbb{R}^{n \times 3}\)</span>, where the first column is the all ones vector for the intercept and the second and third columns are <span class="math inline">\({\bf X} \sim \text{N}(0,1)\)</span> and <span class="math inline">\({\bf E} \sim \text{N}(0,1)\)</span>, respectively. The last two columns are scaled to have <span class="math inline">\(0\)</span> mean and unit variance.</p>
<p>The <code>makeSimData</code> function additionally generates the genetic marker matrix <span class="math inline">\({\bf G}\)</span> with synthetic haplotype data from the COSI software. Detailed procedures for generating <span class="math inline">\({\bf G}\)</span> can be found in the accompanying journal manuscript. Finally, the <code>makeSimData</code> function also generates a continuous phenotype <span class="math inline">\({\bf y}\)</span> according to the following fixed effects model <span class="math display">\[
{\bf y} = \tilde{\bf X} \boldsymbol{\gamma}_{\widetilde{\bf X}} + {\bf G}\boldsymbol{\gamma}_{G} + \text{diag}(E){\bf G}\boldsymbol{\gamma}_{GE} + {\bf e}.
\]</span> Here, <span class="math inline">\(\boldsymbol{\gamma}_{\tilde{\bf X}}\)</span> is the all ones vector of length <span class="math inline">\(P=3\)</span>, <span class="math inline">\(\boldsymbol{\gamma}_{G} \in \mathbb{R}^{L}\)</span>, <span class="math inline">\(\boldsymbol{\gamma}_{GE}\in \mathbb{R}^{L}\)</span>, and <span class="math inline">\({\bf e} \sim \text{N}({\bf 0}, \sigma\, {\bf I}_{n})\)</span>. The entries of <span class="math inline">\(\boldsymbol{\gamma}_{G}\)</span> and <span class="math inline">\(\boldsymbol{\gamma}_{GE}\)</span> pertaining to causal loci are set to be <span class="math inline">\(\gamma_{G}\)</span> = <code>gammaG</code> and <span class="math inline">\(\gamma_{GE}\)</span> = <code>gammaGE</code>, respectively. The remaining entries of <span class="math inline">\(\boldsymbol{\gamma}_{G}\)</span> and <span class="math inline">\(\boldsymbol{\gamma}_{GE}\)</span> pertaining to non-causal loci are set to <span class="math inline">\(0\)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeSimData.html">makeSimData</a></span><span class="op">(</span>H<span class="op">=</span><span class="va">cosihap</span>, n<span class="op">=</span><span class="fl">5000</span>, L<span class="op">=</span><span class="fl">100</span>, gammaG<span class="op">=</span><span class="fl">1</span>, gammaGE<span class="op">=</span><span class="fl">0</span>, causal<span class="op">=</span><span class="fl">40</span>, seed<span class="op">=</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p>Now that we have our data, we can prepare it for use in the SEAGLE algorithm. We will input our <span class="math inline">\({\bf y}\)</span>, <span class="math inline">\({\bf X}\)</span>, <span class="math inline">\({\bf E}\)</span>, and <span class="math inline">\({\bf G}\)</span> into the <code>prep.SEAGLE</code> function. The <code>intercept = 1</code> parameter indicates that the first column of <span class="math inline">\({\bf X}\)</span> is the all ones vector for the intercept.</p>
<p>This preparation procedure formats the input data for the <code>SEAGLE</code> function by checking the dimensions of the input data. It also pre-computes a QR decomposition for <span class="math inline">\(\widetilde{\bf X} = \begin{pmatrix} {\bf 1}_{n} &amp; {\bf X} &amp; {\bf E} \end{pmatrix}\)</span>, where <span class="math inline">\({\bf 1}_{n}\)</span> denotes the all ones vector of length <span class="math inline">\(n\)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">objSEAGLE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep.SEAGLE.html">prep.SEAGLE</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">dat</span><span class="op">$</span><span class="va">y</span>, X<span class="op">=</span><span class="va">dat</span><span class="op">$</span><span class="va">X</span>, intercept<span class="op">=</span><span class="fl">1</span>, E<span class="op">=</span><span class="va">dat</span><span class="op">$</span><span class="va">E</span>, G<span class="op">=</span><span class="va">dat</span><span class="op">$</span><span class="va">G</span><span class="op">)</span></code></pre></div>
<p>Finally, we’ll input the prepared data into the <code>SEAGLE</code> function to compute the score-like test statistic <span class="math inline">\(T\)</span> and its corresponding p-value. The <code>init.tau</code> and <code>init.sigma</code> parameters are the initial values for <span class="math inline">\(\tau\)</span> and <span class="math inline">\(\sigma\)</span> employed in the REML EM algorithm.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SEAGLE.html">SEAGLE</a></span><span class="op">(</span><span class="va">objSEAGLE</span>, init.tau<span class="op">=</span><span class="fl">0.5</span>, init.sigma<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>
<span class="va">res</span><span class="op">$</span><span class="cn">T</span>
<span class="co">#&gt; [1] 246.1886</span>
<span class="va">res</span><span class="op">$</span><span class="va">pv</span>
<span class="co">#&gt; [1] 0.8441451</span></code></pre></div>
<p>The score-like test statistic <span class="math inline">\(T\)</span> for the G<span class="math inline">\(\times\)</span>E effect and its corresponding p-value can be found in <code>res$T</code> and <code>res$pv</code>, respectively.</p>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="http://jocelynchi.com">Jocelyn Chi</a>, <a href="https://ipsen.math.ncsu.edu/">Ilse Ipsen</a>, <a href="https://www4.stat.ncsu.edu/~jytzeng/">Jung-Ying Tzeng</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
