<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example 3: Using SEAGLE with GWAS or Next Generation Sequencing Data • SEAGLE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example 3: Using SEAGLE with GWAS or Next Generation Sequencing Data">
<meta property="og:description" content="SEAGLE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SEAGLE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/example1.html">Example 1: Using SEAGLE with .txt Input Files</a>
    </li>
    <li>
      <a href="../articles/example2.html">Example 2: Using SEAGLE with Simulated Data</a>
    </li>
    <li>
      <a href="../articles/example3.html">Example 3: Using SEAGLE with GWAS or Next Generation Sequencing Data</a>
    </li>
    <li>
      <a href="../articles/example4.html">Example 4: Using SEAGLE for Chromosome-Wide Gene-Based Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jocelynchi/SEAGLE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="example3_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example 3: Using SEAGLE with GWAS or Next Generation Sequencing Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jocelynchi/SEAGLE/blob/master/vignettes/example3.Rmd"><code>vignettes/example3.Rmd</code></a></small>
      <div class="hidden name"><code>example3.Rmd</code></div>

    </div>

    
    
<p>This tutorial demonstrates how to use the <code>SEAGLE</code> package when the genotype data are from GWAS studies (.bed + .bim + .fam) or next generation sequencing studies (.vcf). We recommend using PLINK1.9 available from <a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a> to generate a .raw file (see <a href="https://www.cog-genomics.org/plink/2.0/formats#raw">https://www.cog-genomics.org/plink/2.0/formats#raw</a>) for each of the SNP sets. The .raw file records the allelic dosage for each SNP.</p>
<p>Examples files containing <code>1kg_phase1_chr22.bed</code>, <code>1kg_phase1_chr22.bim</code>, <code>1kg_phase1_chr22.fam</code>, and <code>1kg_phase1_chr22.vcf</code> files and a corresponding gene list in <code>glist-hg19</code> can be downloaded via the <code>SEAGLE-example3-data.zip</code> file from <a href="http://jocelynchi.com/SEAGLE/SEAGLE-example3-data.zip">http://jocelynchi.com/SEAGLE/SEAGLE-example3-data.zip</a>. To follow along with the PLINK conversion procedures, you will first need to install PLINK1.9 from <a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a> and unzip the example <code>SEAGLE-example3-data.zip</code> file. Afterwards, you can type the PLINK commands below in a command prompt or terminal window in the folder where these files are located. This will produce .raw files that you can read into R to obtain the relevant genetic marker matrix <span class="math inline">\({\bf G}\)</span> for input into <code>SEAGLE</code>.</p>
<div id="preparing-gwas-data-with-plink1-9" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-gwas-data-with-plink1-9" class="anchor"></a>Preparing GWAS Data with PLINK1.9</h2>
<p>For GWAS data in the .bed + .bim + .fam format, we recommend first converting to the .raw format with PLINK1.9. To illustrate the analysis of a SNP set from a single gene, we employ the gene ACR as an example. We type the following commands in command prompt or terminal to produce the corresponding .raw file, which includes all SNPs from gene ACR.</p>
<pre><code>plink --bfile 1kg_phase1_chr22 --make-set glist-hg19 --gene ACR --out ACR --export A</code></pre>
<p>To illustrate the analysis of a SNP set from multiple genes, we employ the sequence of genes from A4GALT and ACR as an example. We type the following commands in command prompt or terminal to produce the corresponding .raw file, which includes all SNPs from genes A4GALT and ACR.</p>
<pre><code>plink --bfile 1kg_phase1_chr22 --make-set glist-hg19 --gene A4GALT ACR --out A4GALT_ACR --export A</code></pre>
</div>
<div id="preparing-next-generation-sequencing-data-with-plink1-9" class="section level2">
<h2 class="hasAnchor">
<a href="#preparing-next-generation-sequencing-data-with-plink1-9" class="anchor"></a>Preparing Next Generation Sequencing Data with PLINK1.9</h2>
<p>For next generation sequencing data in the .vcf format, we recommend first converting to the .raw format with PLINK1.9. To illustrate the analysis of a SNP set from a single gene, we employ the gene ACR as an example. We type the following commands in command prompt or terminal to produce the corresponding .raw file, which includes all SNPs from gene ACR.</p>
<pre><code>plink --vcf 1kg_phase1_chr22.vcf --make-set glist-hg19 --gene ACR --out ACR --export A</code></pre>
<p>To illustrate the analysis of a SNP set from multiple genes, we employ the sequence of genes from A4GALT and ACR as an example. We type the following commands in command prompt or terminal to produce the corresponding .raw file, which includes all SNPs from genes A4GALT and ACR.</p>
<pre><code>plink --vcf 1kg_phase1_chr22.vcf --make-set glist-hg19 --gene A4GALT ACR --out A4GALT_ACR --export A</code></pre>
</div>
<div id="loading--raw-files-into-r-and-extracting-the-genetic-marker-matrix-bf-g" class="section level2">
<h2 class="hasAnchor">
<a href="#loading--raw-files-into-r-and-extracting-the-genetic-marker-matrix-bf-g" class="anchor"></a>Loading .raw files into R and extracting the genetic marker matrix <span class="math inline">\({\bf G}\)</span>
</h2>
<p>Now that we have the genotype data in the .raw format, we can load it into R as follows. We’ll first load the <code>SEAGLE</code> package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jocelynchi/SEAGLE">SEAGLE</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Loading required package: CompQuadForm</span></code></pre></div>
<p>The following code will load the <code>ACR.raw</code> file produced from the PLINK conversion for GWAS data for single gene analysis described above. Of course, the actual file location will depend on where you stored the results of the PLINK conversion on your local drive.</p>
<pre><code>acr &lt;- read.delim("ACR.raw", sep=" ")</code></pre>
<p>As an example, we’ve included the <code>ACR.raw</code> file in the <code>extdata</code> folder of this package. The following code loads that file so we can use it in this tutorial.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">acr_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ACR.raw"</span>, package <span class="op">=</span> <span class="st">"SEAGLE"</span><span class="op">)</span>
<span class="va">acr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">acr_loc</span>, sep<span class="op">=</span><span class="st">" "</span><span class="op">)</span></code></pre></div>
<p>We can take a quick look at the resulting <code>acr</code> object. The first 6 columns correspond to identifying information.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">acr</span><span class="op">)</span>
<span class="co">#&gt; [1] 1092  107</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">acr</span><span class="op">)</span>
<span class="co">#&gt;   FID     IID PAT MAT SEX PHENOTYPE rs200755264_A rs145373249_G rs201146734_G</span>
<span class="co">#&gt; 1   0 HG00096   0   0   1        -9             0             0             0</span>
<span class="co">#&gt; 2   0 HG00097   0   0   2        -9             0             0             0</span>
<span class="co">#&gt; 3   0 HG00099   0   0   2        -9             0             0             0</span>
<span class="co">#&gt; 4   0 HG00100   0   0   2        -9             0             0             0</span>
<span class="co">#&gt; 5   0 HG00101   0   0   1        -9             0             0             0</span>
<span class="co">#&gt; 6   0 HG00102   0   0   2        -9             0             0             0</span>
<span class="co">#&gt;   rs79314756_T rs200079264_C rs142494818_T rs77216309_T rs190253671_C</span>
<span class="co">#&gt; 1            0             0             0            0             0</span>
<span class="co">#&gt; 2            0             0             0            0             0</span>
<span class="co">#&gt; 3            0             0             0            0             0</span>
<span class="co">#&gt; 4            0             0             0            0             0</span>
<span class="co">#&gt; 5            0             0             0            0             0</span>
<span class="co">#&gt; 6            0             0             0            0             0</span>
<span class="co">#&gt;   rs182163802_A rs73174437_T rs12165648_A rs116026001_G rs146910789_T</span>
<span class="co">#&gt; 1             0            0            0             0             0</span>
<span class="co">#&gt; 2             0            0            0             0             0</span>
<span class="co">#&gt; 3             0            0            0             0             0</span>
<span class="co">#&gt; 4             0            0            0             0             0</span>
<span class="co">#&gt; 5             0            1            0             0             0</span>
<span class="co">#&gt; 6             0            0            0             0             0</span>
<span class="co">#&gt;   rs185375060_A rs191165317_A rs140628153_T rs1064733_T rs114634361_T</span>
<span class="co">#&gt; 1             0             0             0           0             0</span>
<span class="co">#&gt; 2             0             0             0           0             0</span>
<span class="co">#&gt; 3             0             0             0           0             0</span>
<span class="co">#&gt; 4             0             0             0           0             0</span>
<span class="co">#&gt; 5             0             0             0           0             0</span>
<span class="co">#&gt; 6             0             0             0           0             0</span>
<span class="co">#&gt;   rs150273293_T rs77313877_A rs189214490_A rs75385660_C rs147948077_A</span>
<span class="co">#&gt; 1             0            0             0            0             0</span>
<span class="co">#&gt; 2             0            0             0            0             0</span>
<span class="co">#&gt; 3             0            0             0            0             0</span>
<span class="co">#&gt; 4             0            0             0            0             0</span>
<span class="co">#&gt; 5             0            0             0            0             0</span>
<span class="co">#&gt; 6             0            0             0            0             0</span>
<span class="co">#&gt;   rs200217266_G rs2285395_A rs143887109_A rs140364806_T rs182889325_T</span>
<span class="co">#&gt; 1             0           0             0             0             0</span>
<span class="co">#&gt; 2             0           0             0             0             0</span>
<span class="co">#&gt; 3             0           0             0             0             0</span>
<span class="co">#&gt; 4             0           0             0             0             0</span>
<span class="co">#&gt; 5             0           0             0             0             0</span>
<span class="co">#&gt; 6             0           0             0             0             0</span>
<span class="co">#&gt;   rs6010067_C rs58109483_G rs141844965_T rs145800154_C rs148192955_T</span>
<span class="co">#&gt; 1           0            0             0             0             0</span>
<span class="co">#&gt; 2           0            0             0             0             0</span>
<span class="co">#&gt; 3           0            0             0             0             0</span>
<span class="co">#&gt; 4           0            0             0             0             0</span>
<span class="co">#&gt; 5           0            0             0             0             0</span>
<span class="co">#&gt; 6           0            0             0             0             0</span>
<span class="co">#&gt;   rs2283675_A rs140237069_C rs191736410_C rs59998603_T rs142530125_G</span>
<span class="co">#&gt; 1           1             0             0            0             0</span>
<span class="co">#&gt; 2           1             0             0            0             0</span>
<span class="co">#&gt; 3           0             0             0            0             0</span>
<span class="co">#&gt; 4           0             0             0            0             0</span>
<span class="co">#&gt; 5           0             0             0            0             0</span>
<span class="co">#&gt; 6           0             0             0            0             0</span>
<span class="co">#&gt;   rs184104408_T rs188103638_T rs146034851_C rs138777072_A rs192800892_T</span>
<span class="co">#&gt; 1             0             0             0             0             0</span>
<span class="co">#&gt; 2             0             0             0             0             0</span>
<span class="co">#&gt; 3             0             0             0             0             0</span>
<span class="co">#&gt; 4             0             0             0             0             0</span>
<span class="co">#&gt; 5             0             0             0             0             0</span>
<span class="co">#&gt; 6             0             0             0             0             0</span>
<span class="co">#&gt;   rs2283676_A rs185550366_T rs190654794_C rs181504904_G rs186157598_A</span>
<span class="co">#&gt; 1           1             0             0             0             0</span>
<span class="co">#&gt; 2           1             0             0             0             0</span>
<span class="co">#&gt; 3           0             0             0             0             0</span>
<span class="co">#&gt; 4           0             0             0             0             0</span>
<span class="co">#&gt; 5           0             0             0             0             0</span>
<span class="co">#&gt; 6           0             0             0             0             0</span>
<span class="co">#&gt;   rs188568125_T rs181184416_T rs185832541_A rs5770999_T rs73433406_T</span>
<span class="co">#&gt; 1             0             0             0           1            0</span>
<span class="co">#&gt; 2             0             0             0           2            0</span>
<span class="co">#&gt; 3             0             0             0           0            0</span>
<span class="co">#&gt; 4             0             0             0           0            0</span>
<span class="co">#&gt; 5             0             0             0           0            0</span>
<span class="co">#&gt; 6             0             0             0           0            0</span>
<span class="co">#&gt;   rs150628453_A rs6010068_T rs190741616_T rs6010069_G rs9616824_T rs6010070_G</span>
<span class="co">#&gt; 1             0           1             0           0           1           0</span>
<span class="co">#&gt; 2             0           1             0           0           0           0</span>
<span class="co">#&gt; 3             0           1             0           1           0           1</span>
<span class="co">#&gt; 4             0           0             0           1           0           0</span>
<span class="co">#&gt; 5             0           1             0           1           0           1</span>
<span class="co">#&gt; 6             0           0             0           0           0           0</span>
<span class="co">#&gt;   rs183029053_G rs150059289_C rs186986450_T rs201638160_T rs57763389_A</span>
<span class="co">#&gt; 1             0             0             0             0            0</span>
<span class="co">#&gt; 2             0             0             0             0            0</span>
<span class="co">#&gt; 3             0             0             0             0            0</span>
<span class="co">#&gt; 4             0             0             0             0            0</span>
<span class="co">#&gt; 5             0             0             0             0            0</span>
<span class="co">#&gt; 6             0             0             0             0            0</span>
<span class="co">#&gt;   rs144808185_T rs138662706_C rs199878967_A rs141891250_T rs201174415_CT</span>
<span class="co">#&gt; 1             0             0             0             0              0</span>
<span class="co">#&gt; 2             0             0             0             0              0</span>
<span class="co">#&gt; 3             0             0             0             0              0</span>
<span class="co">#&gt; 4             0             0             0             0              0</span>
<span class="co">#&gt; 5             1             1             1             0              0</span>
<span class="co">#&gt; 6             1             1             1             0              0</span>
<span class="co">#&gt;   rs112131405_C rs13058392_G rs149554062_A rs6010072_G rs9616953_C rs6009960_G</span>
<span class="co">#&gt; 1             0            0             0           0           0           0</span>
<span class="co">#&gt; 2             0            0             0           0           0           0</span>
<span class="co">#&gt; 3             1            0             1           1           1           1</span>
<span class="co">#&gt; 4             1            0             1           1           1           1</span>
<span class="co">#&gt; 5             1            0             0           1           1           1</span>
<span class="co">#&gt; 6             1            0             0           1           1           0</span>
<span class="co">#&gt;   rs9616954_G rs13056271_C rs56807126_C rs34756634_G rs9616955_T rs13056621_A</span>
<span class="co">#&gt; 1           0            0            0            0           0            0</span>
<span class="co">#&gt; 2           0            0            0            1           0            1</span>
<span class="co">#&gt; 3           1            0            0            1           1            1</span>
<span class="co">#&gt; 4           0            1            1            0           0            0</span>
<span class="co">#&gt; 5           0            1            0            0           0            0</span>
<span class="co">#&gt; 6           0            0            0            0           0            1</span>
<span class="co">#&gt;   rs201458351_G rs9616825_G rs182754197_G rs146416000_A rs6010073_C</span>
<span class="co">#&gt; 1             0           1             0             0           0</span>
<span class="co">#&gt; 2             0           2             1             0           0</span>
<span class="co">#&gt; 3             0           0             0             0           2</span>
<span class="co">#&gt; 4             0           0             0             0           0</span>
<span class="co">#&gt; 5             0           0             0             0           1</span>
<span class="co">#&gt; 6             1           0             0             0           0</span>
<span class="co">#&gt;   rs187441595_T rs116139503_G rs190644757_G rs28516879_A rs182503014_T</span>
<span class="co">#&gt; 1             0             0             0            0             0</span>
<span class="co">#&gt; 2             0             0             0            0             1</span>
<span class="co">#&gt; 3             0             0             0            1             0</span>
<span class="co">#&gt; 4             0             0             0            0             0</span>
<span class="co">#&gt; 5             0             0             0            1             1</span>
<span class="co">#&gt; 6             0             0             0            0             1</span>
<span class="co">#&gt;   rs188459010_C rs56238942_G rs6009961_A rs6010074_T rs139027620_C rs58651371_T</span>
<span class="co">#&gt; 1             0            0           1           0             0            1</span>
<span class="co">#&gt; 2             0            0           2           0             0            2</span>
<span class="co">#&gt; 3             0            0           0           0             1            0</span>
<span class="co">#&gt; 4             0            0           0           0             0            0</span>
<span class="co">#&gt; 5             0            0           0           0             1            1</span>
<span class="co">#&gt; 6             0            0           0           0             1            1</span>
<span class="co">#&gt;   rs71232632_G rs5771002_A rs184517959_T rs201494682_T</span>
<span class="co">#&gt; 1            0           1             0             0</span>
<span class="co">#&gt; 2            0           1             0             0</span>
<span class="co">#&gt; 3            0           1             2             0</span>
<span class="co">#&gt; 4            0           1             0             0</span>
<span class="co">#&gt; 5            0           1             1             0</span>
<span class="co">#&gt; 6            0           2             0             0</span></code></pre></div>
<p>The remaining columns contain our genetic marker matrix <span class="math inline">\({\bf G}\)</span>. We’ll extract these to input into <code>SEAGLE</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">G</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">acr</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">G</span><span class="op">)</span>
<span class="co">#&gt; [1] 1092  101</span></code></pre></div>
</div>
<div id="running-seagle" class="section level2">
<h2 class="hasAnchor">
<a href="#running-seagle" class="anchor"></a>Running SEAGLE</h2>
<p>As an example, we’ll generate synthetic phenotype, <span class="math inline">\({\bf y}\)</span>, and covariate data, <span class="math inline">\({\bf X}\)</span> and <span class="math inline">\({\bf E}\)</span>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Determine number of individuals and loci</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">G</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">G</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>

<span class="co"># Generate synthetic phenotype and covariate data</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>
<span class="va">X</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>
<span class="va">E</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span></code></pre></div>
<p>Now that we have <span class="math inline">\({\bf y}\)</span>, <span class="math inline">\({\bf X}\)</span>, <span class="math inline">\({\bf E}\)</span>, and <span class="math inline">\({\bf G}\)</span>, we can prepare our data for input into <code>SEAGLE</code>. We will input our <span class="math inline">\({\bf y}\)</span>, <span class="math inline">\({\bf X}\)</span>, <span class="math inline">\({\bf E}\)</span>, and <span class="math inline">\({\bf G}\)</span> into the <code>prep.SEAGLE</code> function. The <code>intercept = 0</code> parameter indicates that the first column of <span class="math inline">\({\bf X}\)</span> is not the all ones vector for the intercept.</p>
<p>This preparation procedure formats the input data for the <code>SEAGLE</code> function by checking the dimensions of the input data. It also pre-computes a QR decomposition for <span class="math inline">\(\widetilde{\bf X} = \begin{pmatrix} {\bf 1}_{n} &amp; {\bf X} &amp; {\bf E} \end{pmatrix}\)</span>, where <span class="math inline">\({\bf 1}_{n}\)</span> denotes the all ones vector of length <span class="math inline">\(n\)</span>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">objSEAGLE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep.SEAGLE.html">prep.SEAGLE</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">y</span>, X<span class="op">=</span><span class="va">X</span>, intercept<span class="op">=</span><span class="fl">0</span>, E<span class="op">=</span><span class="va">E</span>, G<span class="op">=</span><span class="va">G</span><span class="op">)</span></code></pre></div>
<p>Now we can input the prepared data into the <code>SEAGLE</code> function to compute the score-like test statistic <span class="math inline">\(T\)</span> and its corresponding p-value. The <code>init.tau</code> and <code>init.sigma</code> parameters are the initial values for estimating <span class="math inline">\(\tau\)</span> and <span class="math inline">\(\sigma\)</span> in the REML EM algorithm.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SEAGLE.html">SEAGLE</a></span><span class="op">(</span><span class="va">objSEAGLE</span>, init.tau<span class="op">=</span><span class="fl">0.5</span>, init.sigma<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>
<span class="va">res</span><span class="op">$</span><span class="cn">T</span>
<span class="co">#&gt; [1] 915.93</span>
<span class="va">res</span><span class="op">$</span><span class="va">pv</span>
<span class="co">#&gt; [1] 0.725045</span></code></pre></div>
<p>The score-like test statistic <span class="math inline">\(T\)</span> for the G<span class="math inline">\(\times\)</span>E effect and its corresponding p-value can be found in <code>res$T</code> and <code>res$pv</code>, respectively.</p>
</div>
<div id="acknowledgments" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h2>
<p>Many thanks to Yueyang Huang for his help with generating the example data and PLINK1.9 code for this tutorial.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="http://jocelynchi.com">Jocelyn Chi</a>, <a href="https://ipsen.math.ncsu.edu/">Ilse Ipsen</a>, <a href="https://www4.stat.ncsu.edu/~jytzeng/">Jung-Ying Tzeng</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
