<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example 4: Using SEAGLE for Chromosome-Wide Gene-Based Analysis â€¢ SEAGLE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example 4: Using SEAGLE for Chromosome-Wide Gene-Based Analysis">
<meta property="og:description" content="SEAGLE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SEAGLE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/example1.html">Example 1: Using SEAGLE with .txt Input Files</a>
    </li>
    <li>
      <a href="../articles/example2.html">Example 2: Using SEAGLE with Simulated Data</a>
    </li>
    <li>
      <a href="../articles/example3.html">Example 3: Using SEAGLE with GWAS or Next Generation Sequencing Data</a>
    </li>
    <li>
      <a href="../articles/example4.html">Example 4: Using SEAGLE for Chromosome-Wide Gene-Based Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jocelynchi/SEAGLE/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="example4_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example 4: Using SEAGLE for Chromosome-Wide Gene-Based Analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jocelynchi/SEAGLE/blob/master/vignettes/example4.Rmd"><code>vignettes/example4.Rmd</code></a></small>
      <div class="hidden name"><code>example4.Rmd</code></div>

    </div>

    
    
<p>This tutorial demonstrates how to use the <code>SEAGLE</code> package for chromosome-wide gene-based studies when the genotype data are from GWAS studies (.bed + .bim + .fam). We recommend using PLINK1.9 available from <a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a> to generate a .raw file for each of the SNP sets. For more information on .raw files, please refer to <a href="https://www.cog-genomics.org/plink/2.0/formats#raw">https://www.cog-genomics.org/plink/2.0/formats#raw</a>. The .raw file records the allelic dosage for each SNP.</p>
<p>Examples files containing GWAS data (.bed, .bim, and .fam files) for all the genes in chromosome 22 and a corresponding gene list in <code>glist-hg19</code> can be downloaded via the <code>SEAGLE-example4-data.zip</code> file from <a href="http://jocelynchi.com/SEAGLE/SEAGLE-example4-data.zip">http://jocelynchi.com/SEAGLE/SEAGLE-example4-data.zip</a>.</p>
<p>To follow along with the PLINK conversion procedures, you will first need to install PLINK1.9 from <a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a> and unzip the example <code>SEAGLE-example4-data.zip</code> file. Afterwards, you can type the PLINK commands below in a command prompt or terminal window in the folder where these files are located. This will produce a .raw file for each gene that you can read into R to obtain a relevant genetic marker matrix <span class="math inline">\({\bf G}\)</span> that you can input into <code>SEAGLE</code>.</p>
<p>After unzipping the <code>SEAGLE-example4-data.zip</code> file, you will find two empty directories named <code>plink_bash</code> and <code>R_codes</code>. The <code>plink_bash</code> directory is where we will write the PLINK1.9 commands that you will run to create the .raw files for each gene. The <code>R_codes</code> directory is where we will write the R scripts for loading the resulting .raw files into R.</p>
<div id="writing--raw-files-for-each-gene-in-chromosome-22-with-plink1-9" class="section level2">
<h2 class="hasAnchor">
<a href="#writing--raw-files-for-each-gene-in-chromosome-22-with-plink1-9" class="anchor"></a>Writing .raw files for each gene in chromosome 22 with PLINK1.9</h2>
<p>We will begin by telling R the location where you unzipped the <code>SEAGLE-example4-data.zip</code> file. Then we will read in the gene list in <code>glist-hg19</code> and subset for the genes in chromosome 22. If you unzipped the file to your Downloads directory on a Mac, your path might look like the following.</p>
<pre><code>dir &lt;- "/Users/user_name/Downloads/SEAGLE-example4-data/"
glist &lt;- read.table(paste0(dir,"glist-hg19"), header = F)
glist_chr22 &lt;- subset(glist, glist$V1 == 22)</code></pre>
<p>Next, we will remove any duplicated genes for chromosome 22 in our gene list.</p>
<pre><code>dup &lt;- glist_chr22$V4[duplicated(glist_chr22$V4)] # find duplicated gene
dup</code></pre>
<p>Notice that in this example, there are 4 duplicated genes given by the following.</p>
<pre><code># GSTT2: chr22:24,322,339-24,326,106(GRCh37/hg19 by Ensembl)
# GSTT2B: chr22:24,299,601-24,303,373(GRCh37/hg19 by Ensembl)
# RIMBP3B: chr22:21,738,040-21,743,458(GRCh37/hg19 by Entrez Gene)
# RIMBP3C: chr22:21,899,646-21,905,750(GRCh37/hg19 by Ensembl)</code></pre>
<p>We will update our gene list by manually updating the positions of the duplicated genes.</p>
<pre><code>glist_chr22 &lt;- glist_chr22[!duplicated(glist_chr22$V4),]
glist_chr22[which(glist_chr22$V4 == "GSTT2"),2:3] &lt;- c(24322339, 24326106)
glist_chr22[which(glist_chr22$V4 == "GSTT2B"),2:3] &lt;- c(24299601,24303373)
glist_chr22[which(glist_chr22$V4 == "RIMBP3B"),2:3] &lt;- c(21738040,21743458)
glist_chr22[which(glist_chr22$V4 == "RIMBP3C"),2:3] &lt;- c(21899646,21905750)
write.table(glist_chr22, "glist-hg19_chr22_updated", col.names = F, row.names = F, quote = F)

# replace hyphen with underscore
glist_chr22$V4 &lt;- sub("-", "_", glist_chr22$V4)</code></pre>
<p>Now we will use PLINK1.9 to create a .raw file for each gene in chromosome 22, which includes all SNPs for a given gene. The following code will produce a sequence of bash files to produce .raw files for each gene in <code>glist_chr22</code>. It will also produce a corresponding sequence of R code snippets for reaching each .raw file into R. Each bash file will run PLINK1.9 for <code>num_genes=50</code> genes at a time.</p>
<pre><code># of plink commands per job
num_genes &lt;- 50

bash_plink &lt;- NULL # bash command
command &lt;- c("#! /bin/bash", "#SBATCH --job-name=plink") # plink command
codeR &lt;- NULL # R code
for (i in 1:nrow(glist_chr22)) {
  command &lt;- c(command, paste0("plink --bfile 1kg_phase1_chr22 --make-set
  glist-hg19_chr22_updated --gene ", glist_chr22$V4[i]," --out ",
  glist_chr22$V4[i]," --export A") )

  # write R codes for reading .raw files, one R file for one gene
  codeR &lt;- paste0(glist_chr22$V4[i], " &lt;- read.delim(\"", glist_chr22$V4[i],".raw\", sep=\" \")")
  fileConn&lt;-file(paste0(dir,"R_codes/", glist_chr22$V4[i],".R"))
  writeLines(codeR, fileConn)
  close(fileConn)

  if(i %% num_genes == 0 | i==nrow(glist_chr22)){
    # write plink commands in bash file
    fileConn&lt;-file(paste0(dir,"plink_bash/plink_job",i-num_genes+1,"-",i,".sh"))
    writeLines(command, fileConn)
    close(fileConn)

    command &lt;- c("#! /bin/bash", "#SBATCH --job-name=plink")
    bash_plink &lt;- c(bash_plink, paste0("sbatch plink_job",i-num_genes+1,"-",i,".sh"))
  }
}

# Write bash commands to a text file
fileConn&lt;-file(paste0("bash_plink.txt"))
writeLines(bash_plink, fileConn)
close(fileConn)</code></pre>
<p>Note that if you want to run the .sh files in the <code>plink_bash</code> directory on your local computer, you can employ the <code>bash</code> command instead of <code>SBATCH</code> in terminal or command prompt. Additionally, if you are using Linux, you may need to specify <code>plink1.9</code> rather than just <code>plink</code>.</p>
<p>Running the above code will setup the code and files to produce a single .raw file for each gene when running the commands in the <code>bash_plink.txt</code> file. The .raw file for each gene will contain all the SNPs for that gene.</p>
</div>
<div id="loading--raw-files-into-r-and-extracting-the-genetic-marker-matrix-bf-g" class="section level2">
<h2 class="hasAnchor">
<a href="#loading--raw-files-into-r-and-extracting-the-genetic-marker-matrix-bf-g" class="anchor"></a>Loading .raw files into R and extracting the genetic marker matrix <span class="math inline">\({\bf G}\)</span>
</h2>
<p>Now that we have data for each gene in the .raw format, we can load the .raw files sequentially into R for input into <code>SEAGLE</code> as follows. Weâ€™ll first load the <code>SEAGLE</code> package.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jocelynchi/SEAGLE">SEAGLE</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Loading required package: CompQuadForm</span></code></pre></div>
<p>The following code shows how to test for the GxE interaction effect using <code>SEAGLE</code> on these genes. To illustrate how to use SEAGLE, we will generate synthetic phenotype and covariate data for <span class="math inline">\(n=1092\)</span> study participants.</p>
<p>As an example, weâ€™ve included .raw files for the following subset of genes from chromosome 22 in the <code>extdata</code> folder of this package: ACR, APOBEC3A, APOBEC3C, ARSA, ATF4, ATP5L2, BCRP2, BMS1P17, and BMS1P18. Weâ€™ve also included a corresponding gene list in <code>glist-hg19_chr22_example</code>. In practice, you will want to specify your own <code>dir</code> for the directory where youâ€™ve stored your .raw files.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read in gene list</span>
<span class="va">dir</span> <span class="op">&lt;-</span> <span class="st">"../inst/extdata/"</span>
<span class="va">genelist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"glist-hg19_chr22_example"</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">" "</span>, header<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Identify number of genes in genelist</span>
<span class="va">num_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">genelist</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

<span class="co"># Generate synthetic phenotype and covariate data</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">1092</span> <span class="co"># number of study participants</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>
<span class="va">X</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>
<span class="va">E</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Now that we have <span class="math inline">\({\bf y}\)</span>, <span class="math inline">\({\bf X}\)</span>, <span class="math inline">\({\bf E}\)</span>, and <span class="math inline">\({\bf G}\)</span>, we can run <code>SEAGLE</code> on each gene in <code>genelist</code>. We will first perform data checking procedures on <span class="math inline">\({\bf G}\)</span>. Then we will input <span class="math inline">\({\bf y}\)</span>, <span class="math inline">\({\bf X}\)</span>, <span class="math inline">\({\bf E}\)</span>, and <span class="math inline">\({\bf G}\)</span> for each gene into the <code>prep.SEAGLE</code> function. The <code>intercept = 0</code> parameter indicates that the first column of <span class="math inline">\({\bf X}\)</span> is not the all ones vector for the intercept.</p>
<p>The preparation procedure formats the input data for the <code>SEAGLE</code> function by checking the dimensions of the input data. It also pre-computes a QR decomposition for <span class="math inline">\(\widetilde{\bf X} = \begin{pmatrix} {\bf 1}_{n} &amp; {\bf X} &amp; {\bf E} \end{pmatrix}\)</span>, where <span class="math inline">\({\bf 1}_{n}\)</span> denotes the all ones vector of length <span class="math inline">\(n\)</span>.</p>
<p>Afterwards, we will input the prepared data for each gene into the <code>SEAGLE</code> function to compute the score-like test statistic <span class="math inline">\(T\)</span> and its corresponding p-value. The <code>init.tau</code> and <code>init.sigma</code> parameters are the initial values for estimating <span class="math inline">\(\tau\)</span> and <span class="math inline">\(\sigma\)</span> in the REML EM algorithm.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Initialize output containers for T and p-value for each gene</span>
<span class="va">T.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">numeric</a></span><span class="op">(</span>length<span class="op">=</span><span class="va">num_genes</span><span class="op">)</span>
<span class="va">pv.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">numeric</a></span><span class="op">(</span>length<span class="op">=</span><span class="va">num_genes</span><span class="op">)</span>

<span class="co"># Run SEAGLE on each gene in genelist</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">num_genes</span><span class="op">)</span> <span class="op">{</span>
  
  <span class="co"># Identify current gene</span>
  <span class="va">gene_name</span> <span class="op">&lt;-</span> <span class="va">genelist</span><span class="op">[</span><span class="va">i</span>,<span class="fl">4</span><span class="op">]</span>
  
  <span class="co"># Obtain G</span>
  <span class="va">Gtemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">dir</span>, <span class="va">gene_name</span>, <span class="st">".raw"</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">" "</span><span class="op">)</span>
  <span class="va">G</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">Gtemp</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
  <span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">G</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>                <span class="co">## number of SNPs</span>
  
  <span class="co"># Make weights</span>
  <span class="va">avg_newsnp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span><span class="op">(</span><span class="va">G</span><span class="op">)</span>
  <span class="va">fA</span>  <span class="op">=</span> <span class="va">avg_newsnp</span><span class="op">/</span><span class="fl">2</span>            <span class="co">## freq of allele "A"</span>
  <span class="va">fa</span>  <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">fA</span>                    <span class="co">## freq of allele "a"</span>
  <span class="va">maf</span> <span class="op">=</span> <span class="va">fA</span>; <span class="va">maf</span><span class="op">[</span><span class="va">fA</span><span class="op">&gt;</span><span class="fl">0.5</span><span class="op">]</span><span class="op">=</span> <span class="va">fa</span><span class="op">[</span><span class="va">fA</span><span class="op">&gt;</span><span class="fl">0.5</span><span class="op">]</span><span class="co">## maf should be b/w 0 and 0.5</span>
  <span class="va">G</span> <span class="op">=</span> <span class="va">G</span><span class="op">[</span> ,<span class="va">maf</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">]</span>                <span class="co">## only keep those SNPs with MAF&gt;0</span>
  <span class="va">maf</span> <span class="op">&lt;-</span> <span class="va">maf</span><span class="op">[</span><span class="va">maf</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span>           <span class="co">## Keep only MAF &gt; 0)</span>
  <span class="va">wt</span>   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">maf</span><span class="op">^</span><span class="op">(</span><span class="op">-</span><span class="fl">3</span><span class="op">/</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>       <span class="co">## Note we take the square root here</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">wt</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">G_final</span>    <span class="op">=</span> <span class="va">G</span> <span class="op">%*%</span> <span class="fu"><a href="https://rdrr.io/r/base/diag.html">diag</a></span><span class="op">(</span><span class="va">wt</span><span class="op">)</span> <span class="co">## Input this G</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">T.list</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="va">pv.list</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
    <span class="kw">next</span>
  <span class="op">}</span>

  <span class="co"># Run SEAGLE</span>
  <span class="va">objSEAGLE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prep.SEAGLE.html">prep.SEAGLE</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">y</span>, X<span class="op">=</span><span class="va">X</span>, intercept<span class="op">=</span><span class="fl">0</span>, 
                           E<span class="op">=</span><span class="va">E</span>, G<span class="op">=</span><span class="va">G_final</span><span class="op">)</span>
  <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SEAGLE.html">SEAGLE</a></span><span class="op">(</span><span class="va">objSEAGLE</span>, init.tau<span class="op">=</span><span class="fl">0.5</span>, init.sigma<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>
  
  <span class="co"># Save T and p-value into output lists</span>
  <span class="va">T.list</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="cn">T</span>
  <span class="va">pv.list</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">pv</span>
<span class="op">}</span></code></pre></div>
<p>The score-like test statistics <span class="math inline">\(T\)</span> for the G<span class="math inline">\(\times\)</span>E effect and their corresponding p-values can be found in <code>T.list</code> and <code>pv.list</code>, respectively. We can take a look at the test statistics and p-values computed for each of the genes in <code>genelist</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">resMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">T.list</span>, <span class="va">pv.list</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">resMat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"T"</span>, <span class="st">"p-value"</span><span class="op">)</span>
<span class="va">resMat</span>
<span class="co">#&gt;               T   p-value</span>
<span class="co">#&gt;  [1,] 7693.9187 0.6148780</span>
<span class="co">#&gt;  [2,] 4022.4392 0.7990406</span>
<span class="co">#&gt;  [3,] 3375.4804 0.9771967</span>
<span class="co">#&gt;  [4,] 9698.0070 0.2089241</span>
<span class="co">#&gt;  [5,] 3523.1752 0.3668638</span>
<span class="co">#&gt;  [6,]  716.1017 0.3435655</span>
<span class="co">#&gt;  [7,] 4792.4809 0.9336786</span>
<span class="co">#&gt;  [8,]  139.5526 0.9186002</span>
<span class="co">#&gt;  [9,]  139.5526 0.9186002</span></code></pre></div>
</div>
<div id="acknowledgments" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h2>
<p>Many thanks to Yueyang Huang for generating the example data, PLINK1.9 code and bash files, and R scripts for reading in the .raw files for this tutorial.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="http://jocelynchi.com">Jocelyn Chi</a>, <a href="https://ipsen.math.ncsu.edu/">Ilse Ipsen</a>, <a href="https://www4.stat.ncsu.edu/~jytzeng/">Jung-Ying Tzeng</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
